pkg/controllers/externalsecret/externalsecret_controller_test.go

```diff
@@ -2023,6 +2023,45 @@ var _ = Describe("ExternalSecret controller", Serial, func() {
 		}
 	}
 
+	// Secret is created when ClusterSecretStore has a single matching "all" glob condition
```

Comment:
externalsecret_controller_test.go is for the external secret controller. The regex feature is for the cluster secret store controller, which is different from the external secret controller. Would you add tests to pkg/controllers/secretstore/common_test.go? Or maybe pkg/controllers/secretstore/client_manager_test.go is easier to write those tests.

Reply:
Actually, testing it there is super inconvenient. :D

Reply:
Nothing tests `shouldProcessSecret` function in that package. :P

Reply:
I'll write a very simple test.